pkg/virtualnode/enqueue_requests_for_pod_events.go

```diff
@@ -0,0 +1,95 @@
+package virtualnode
+
+import (
+	"context"
+	appmesh "github.com/aws/aws-app-mesh-controller-for-k8s/apis/appmesh/v1beta2"
+	"github.com/aws/aws-app-mesh-controller-for-k8s/pkg/k8s"
+	"github.com/go-logr/logr"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/client-go/util/workqueue"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+)
+
+func NewEnqueueRequestsForPodEvents(k8sClient client.Client, log logr.Logger) *enqueueRequestsForPodEvents {
+	return &enqueueRequestsForPodEvents{
+		k8sClient: k8sClient,
+		log:       log,
+	}
+}
+
+var _ handler.EventHandler = (*enqueueRequestsForPodEvents)(nil)
+
+type enqueueRequestsForPodEvents struct {
+	k8sClient client.Client
+	log       logr.Logger
+}
+
+// Create is called in response to an create event
+func (h *enqueueRequestsForPodEvents) Create(e event.CreateEvent, queue workqueue.RateLimitingInterface) {
+	h.enqueueVirtualNodesForPods(context.Background(), queue, e.Object.(*corev1.Pod))
+}
+
+// Update is called in response to an update event
+func (h *enqueueRequestsForPodEvents) Update(e event.UpdateEvent, queue workqueue.RateLimitingInterface) {
+	// cloudmap reconcile depends Virtualnode Pod Selector labels and if there is an update to a Pod
+	// we need to check if there is any change w.r.t VirtualNode it belongs.
+	oldPod := e.ObjectOld.(*corev1.Pod)
+	newPod := e.ObjectNew.(*corev1.Pod)
+
+	oldPodIsReady := false
+	newPodIsReady := false
+
+	conditions := (&oldPod.Status).Conditions
+	for i := range conditions {
```

Comment:
better to have utilities to test whether pod is ready, and reuse it here.

Reply:
ok